pyam/core.py

```diff
@@ -906,12 +907,15 @@ def check_aggregate_region(self, variable, region='World', subregions=None,
             col_args = dict(region=region, variable=variable)
             return IamDataFrame(diff, **col_args).timeseries()
 
-    def _variable_components(self, variable):
-        """Get all components (sub-categories) of a variable
+    def _variable_components(self, variable, all=False):
+        """Get all components (sub-categories) of a variable for a given level
 
-        For `variable='foo'`, return `['foo|bar']`, but don't include
-        `'foo|bar|baz'`, which is a sub-sub-category"""
+        If `all`, for `variable='foo'`, return `['foo|bar']`, but don't include
```

Comment:
Isn't this exactly the other way round? `all is True` returns all sub-sub-...-categories...